src/main/java/com/visenze/common/util/ViJsonAny.java

```diff
@@ -0,0 +1,155 @@
+package com.visenze.common.util;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.visenze.visearch.internal.InternalViSearchException;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <h1> ViJsonAny - A JsonNode wrapper </h1>
+ * This class is created to handle multiple typed JSON objects that do not have
+ * nested/children json tree nodes. "Multiple typed JSON objects" are those that
+ * hope to be deserialized as:
+ *      Map<String, String OR List<String> OR Map<String, String>>
+ *
+ * The statement above is only possible in Java if it is denoted as:
+ *      Map<String, Object>, or with Json support, Map<String, JsonNode>
+ *
+ * The initial intention is to be a replacement for Map<String, JsonNode>:
+ *      Map<String, ViJsonAny>
+ *
+ * With this wrapper class, you could check if it is a definite value or an
+ * array before extracting it as that type. The same functionality can be
+ * achieved by using JsonNode directly but this class hides the boiler plate
+ * code that you would otherwise need to type.
+ * <p>
+ * It does not support recursively identifying and extracting json nodes' data.
+ *
+ * @author Shannon Tan
+ * @version 1.0
+ * @since 20 Jan 2021
+ */
+public class ViJsonAny {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
```

Comment:
can put this in a common place e.g. abstract class, if there are multiple classes that need to use this. This object is fairly expensive and should be re-used if available.